xclim/core/units.py

```diff
@@ -581,19 +582,32 @@ def to_agg_units(
     >>> degdays.units
     'K d'
     """
-    m, freq_u_raw = infer_sampling_units(orig[dim])
-    freq_u = str2pint(freq_u_raw)
-    orig_u = str2pint(orig.units)
-
-    out = out * m
-    if op == "count":
-        out.attrs["units"] = freq_u_raw
-    elif op == "prod":
-        out.attrs["units"] = pint2cfunits(orig_u * freq_u)
-    elif op == "delta_prod":
-        out.attrs["units"] = pint2cfunits((orig_u - orig_u) * freq_u)
+    if op in ["min", "max", "mean"]:
+        out.attrs["units"] = orig.attrs["units"]
+
+    elif op in ["doymin", "doymax"]:
+        out.attrs.update(
+            units="", is_dayofyear=np.int32(1), calendar=get_calendar(orig)
+        )
+
+    elif op in ["count", "prod", "delta_prod"]:
+        m, freq_u_raw = infer_sampling_units(orig[dim])
```

Comment:
Was not sure what you meant by `doy`, but I implemented the rest of functions you suggested @aulemahal

Reply:
By "doy" I meant a generic "op" name for all operations that return a doy. Like season starts or ends. So that one day, those functions could make use of `to_agg_units`, and we would really centralized all unit-inferring to a single method.